### PR TITLE
add ilm policiy to apm data streams

### PR DESCRIPTION
**Pull request by jalvz**
## Motivation/summary

This copies the current ILM policy to all data streams in the package. 
We should determine if we want to change it for some/all data streams. 
I'm happy to do that as part of this PR or as follow up.

## Checklist

- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

With apm-integration-testing:
`./scripts/compose.py start master --no-download --with-elastic-agent --no-apm-server --with-package-registry --package-registry-apm-path <path-to>/apm-server/apmpackage/apm` install the APM integration and check that the ILM policies for all data streams are installed.

### CODE REVIEW

**Review by simitt — APPROVED** [review]
Let's also wait for @axw but LGTM to start out with the same default policies as we have now.

File: apmpackage/apm/0.1.0/data_stream/app_metrics/elasticsearch/ilm/default_policy.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+                "min_age": "30d",
+                "actions": {
+                    "readonly": {},
+                    "set_priority": {
+                        "priority": 50
+                    }
+                }
+            },
+            "hot": {
+                "min_age": "0ms",
```
Comment (axw) [inline apmpackage/apm/0.1.0/data_stream/app_metrics/elasticsearch/ilm/default_policy.json:14]: Is the min_age needed on the hot phase? I don't think it's normally needed when defining a policy in ES.

**Review by axw — APPROVED** [review]
Yay! Glad to see this added into the package, and looking forward to the day we can remove ILM policy creation from code.

File: apmpackage/apm/0.1.0/data_stream/app_metrics/elasticsearch/ilm/default_policy.json (hunk 1 above)
Comment (simitt) [reply apmpackage/apm/0.1.0/data_stream/app_metrics/elasticsearch/ilm/default_policy.json:14]: good catch - it is not, if the `min_age` is not configured it defaults to `0`.

**Comment by axw** [comment]
Confirmed with BC2 and epr-snapshot, ILM policies are installed by default and associated with data streams:

```
PUT _ilm/policy/traces-apm.traces-default_policy
{
  "policy": {
    "phases": {
      "hot": {
        "min_age": "0ms",
        "actions": {
          "rollover": {
            "max_age": "30d",
            "max_size": "50gb"
          },
          "set_priority": {
            "priority": 100
          }
        }
      },
      "warm": {
        "min_age": "30d",
        "actions": {
          "readonly": {},
          "set_priority": {
            "priority": 50
          }
        }
      }
    }
  }
}
```

traces-apm index template:
```
{
  "index": {
    "lifecycle": {
      "name": "traces-apm.traces-default_policy"
    },
    "codec": "best_compression",
    "mapping": {
      "total_fields": {
        "limit": "10000"
      }
    },
    "refresh_interval": "5s",
    "number_of_shards": "1",
    "query": {
      "default_field": [
        "message"
      ]
    },
    "default_pipeline": "traces-apm-0.1.0-dev.5",
    "number_of_routing_shards": "30"
  }
}
```

**Pull request merged**